airflow-core/tests/unit/api_fastapi/common/db/test_dags.py

```diff
@@ -0,0 +1,273 @@
+        assert dag_row is not None
+        dagrun_state = dag_row[1]
+        assert dagrun_state is not None, "Joined DagRun state must not be None"
+
+    @provide_session
+    def test_includes_queued_run_when_ordering_by_state(
+        self, dag_with_queued_run, dag_with_running_run, session
+    ):
+        """DAGs with QUEUED runs and null start_date, and RUNNING runs must all have joined DagRun info not None."""
+        queued_dag_model, _ = dag_with_queued_run
+        running_dag_model, _ = dag_with_running_run
+
+        query = generate_dag_with_latest_run_query(
+            max_run_filters=[],
+            order_by=SortParam(allowed_attrs=["last_run_state"], model=DagModel).set_value("last_run_state"),
+        )
+        extended_query = query.add_columns(DagRun.state, DagRun.start_date)
+        result = session.execute(extended_query).fetchall()
+
+        # QUEUED DAG
+        queued_row = next((row for row in result if row[0].dag_id == queued_dag_model.dag_id), None)
+        assert queued_row is not None
+        assert queued_row[1] is not None, "Joined DagRun state for QUEUED DAG must not be None"
+        # RUNNING DAG
+        running_row = next((row for row in result if row[0].dag_id == running_dag_model.dag_id), None)
+        assert running_row is not None
+        assert running_row[1] is not None, "Joined DagRun state for RUNNING DAG must not be None"
+        assert running_row[2] is not None, "Joined DagRun start_date for RUNNING DAG must not be None"
+
+    @provide_session
+    def test_includes_queued_run_when_ordering_by_start_date(
+        self, dag_with_queued_run, dag_with_running_run, session
+    ):
+        """DAGs with QUEUED runs and RUNNING runs must all have joined DagRun info not None when ordering by start_date."""
+        queued_dag_model, _ = dag_with_queued_run
+        running_dag_model, _ = dag_with_running_run
+
+        query = generate_dag_with_latest_run_query(
+            max_run_filters=[],
+            order_by=SortParam(allowed_attrs=["last_run_start_date"], model=DagModel).set_value(
+                "last_run_start_date"
+            ),
+        )
+        extended_query = query.add_columns(DagRun.state, DagRun.start_date)
+        result = session.execute(extended_query).fetchall()
+
+        # QUEUED DAG
+        queued_row = next((row for row in result if row[0].dag_id == queued_dag_model.dag_id), None)
+        assert queued_row is not None
+        assert queued_row[1] is not None, "Joined DagRun state for QUEUED DAG must not be None"
+        # RUNNING DAG
+        running_row = next((row for row in result if row[0].dag_id == running_dag_model.dag_id), None)
+        assert running_row is not None
+        assert running_row[1] is not None, "Joined DagRun state for RUNNING DAG must not be None"
+        assert running_row[2] is not None, "Joined DagRun start_date for RUNNING DAG must not be None"
+
+    @provide_session
```

Comment:
```suggestion
```

This is not necessary, `session` is a fixture and will be provided by the test suite.